src/main/java/nl/novi/catsittermanager/services/ImageService.java

```diff
@@ -0,0 +1,136 @@
+
+    @Value("${my.upload_location}")
+    private final Path fileStoragePath;
+    private final String fileStorageLocation;
+    private final FileUploadRepository fileUploadRepository;
+    private final CatRepository catRepository;
+    private final CatsitterRepository catsitterRepository;
+
+    public ImageService(@Value("${my.upload_location}") String fileStorageLocation, FileUploadRepository fileUploadRepository, CatRepository catRepository, CatsitterRepository catsitterRepository) {
+        fileStoragePath = Paths.get(fileStorageLocation).toAbsolutePath().normalize();
+        this.fileStorageLocation = fileStorageLocation;
+        this.fileUploadRepository = fileUploadRepository;
+        this.catRepository = catRepository;
+        this.catsitterRepository = catsitterRepository;
+        try {
+            Files.createDirectories(fileStoragePath);
+        } catch (IOException e) {
+            throw new RuntimeException("Issue in creating file directory");
+        }
+    }
+
+    public String storeFile(MultipartFile file, String url) {
+        String fileName = StringUtils.cleanPath(Objects.requireNonNull(file.getOriginalFilename()));
+        Path filePath = Paths.get(fileStoragePath + "/" + fileName);
+        try {
+            Files.copy(file.getInputStream(), filePath, StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            throw new RuntimeException("Issue in storing the file", e);
+        }
+        fileUploadRepository.save(new FileUploadResponse(fileName, file.getContentType(), url));
+        return fileName;
+    }
+
+    public Resource downloadSingleFile(String fileName) {
+        Path path = Paths.get(fileStorageLocation).toAbsolutePath().resolve(fileName);
+        Resource resource;
+        try {
+            resource = new UrlResource(path.toUri());
+        } catch (MalformedURLException e) {
+            throw new RuntimeException("Issue in reading the file", e);
+        }
+        if (resource.exists() && resource.isReadable()) {
+            return resource;
+        } else {
+            throw new RuntimeException("the file doesn't exist or is not readable");
+        }
+    }
+
+    public FileUploadResponse uploadCatImage(UUID catId, MultipartFile file) {
+        Optional<Cat> optionalCat = catRepository.findById(catId);
+        Cat cat = optionalCat.orElseThrow(() -> new RecordNotFoundException("Cat not found with id: " + catId));
+        String fileName = file.getOriginalFilename();
+        String url = ServletUriComponentsBuilder.fromCurrentContextPath()
+                .path("/cat/")
+                .path(catId.toString())
+                .path("/upload/")
+                .path(fileName)
+                .toUriString();
+        String storedFileName = storeFile(file, url);
+        FileUploadResponse fileUploadResponse = fileUploadRepository.save(new FileUploadResponse(storedFileName, file.getContentType(), url));
+        cat.setImage(fileUploadResponse);
```

Comment:
Hier zit wat overlap in codes. Je gebruikt hier de setter om de image aan de Cat te koppelen. Daarna roep je de methode assignImageToCat waar deze methode ook al in zit verwerkt inclusief het opslaan in de repo.